tests/runtime_tests/test_schema_generation.py

```diff
@@ -688,3 +689,21 @@ def test_json_schema(endpoint):
     # Test that the JSON schema is valid JSON
     json.dumps(schema_inputs)
     json.dumps(schema_outputs)
+
+
+def test_model_config_extra_forbid():
+    class Child(BaseModel):
+        x: str
+        model_config: ConfigDict = ConfigDict(extra="allow")
+
+    class Parent(BaseModel):
+        child: Child
+
+    ApplyParent = apply_function_to_model_tree(
+        Parent,
+        lambda x, y: x,
+        model_kwargs={"model_config": (ConfigDict, ConfigDict(extra="forbid"))},
+    )
+    ApplyChild = ApplyParent.model_fields["child"].annotation
+    assert ApplyChild.model_config["extra"] == "allow"
+    assert ApplyParent.model_config["extra"] == "forbid"
```

Comment:
Can we please try to validate some data here and make sure everything passes / fails as expected? Since we're really close to modifying pydantic internals we want things to be unambiguous.